xbmc/interfaces/json-rpc/FileOperations.cpp

```diff
@@ -164,21 +165,51 @@ JSONRPC_STATUS CFileOperations::GetDirectory(const std::string &method, ITranspo
 JSONRPC_STATUS CFileOperations::GetFileDetails(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
 {
   std::string file = parameterObject["file"].asString();
-  if (!CFile::Exists(file))
-    return InvalidParams;
+  if (!URIUtils::IsPlugin(file))
+  {
+    if (!CFile::Exists(file))
+	  return InvalidParams;
 
-  if (!CFileUtils::RemoteAccessAllowed(file))
-    return InvalidParams;
+	if (!CFileUtils::RemoteAccessAllowed(file))
+	  return InvalidParams;
+  }
+  std::string path;
+  CFileItemList items;
+  CFileItemPtr item;
+  
+  if (!URIUtils::IsPlugin(file))
+  {
+    path = URIUtils::GetDirectory(file);
+    if (path.empty() || !CDirectory::GetDirectory(path, items, "", DIR_FLAG_DEFAULTS) || !items.Contains(file))
+      return InvalidParams;
+    item = items.Get(file);
 
-  std::string path = URIUtils::GetDirectory(file);
+	if (!URIUtils::IsUPnP(file))
```

Comment:
The indentation seems a little off here.

Reply:
I replaces tabs with spaces;
Now it should look better;